br/pkg/restore/split.go

```diff
@@ -125,14 +106,14 @@ func (rs *RegionSplitter) executeSplitByRanges(
 		roughSortedSplitKeys = append(roughSortedSplitKeys, sortedKeys[curRegionIndex])
 	}
 	if len(roughSortedSplitKeys) > 0 {
-		if err := rs.executeSplitByKeys(ctx, splitContext, scanStartKey, roughSortedSplitKeys); err != nil {
```

Comment:
`scanStartKey` is the start key. but `sortedKeys` are all end keys. if didn't scan with start key will cause error.

Reply:
we don't need to split the `scanStartKey` so theoretically we don't need to care about it. For the error I should have fixed in https://github.com/pingcap/tidb/blob/da775f61845ac93544019538bb6f48fb6482d346/br/pkg/restore/split/client.go#L541-L548

Reply:
update: I fixed a bug and add tests in https://github.com/pingcap/tidb/pull/52433/commits/59e6f0f9f3fe14494e18a48c2569f3f76fccba26

Reply:
Okay, We need make sure not use same key as start/end key to scan regions. or we will get an error from PD. I assume there is no change to get a   slice with only one nil/"" value in it.